ws-server/src/server.rs

```diff
@@ -211,67 +191,23 @@ impl Server {
 		subscribe_method_name: &'static str,
 		unsubscribe_method_name: &'static str,
 	) -> Result<SubscriptionSink, Error> {
-		let subscribers = Arc::new(Mutex::new(FxHashMap::default()));
-
-		{
-			let subscribers = subscribers.clone();
-			self.insert_method(
-				subscribe_method_name,
-				Box::new(move |id, _, tx, conn| {
-					let sub_id = {
-						const JS_NUM_MASK: SubscriptionId = !0 >> 11;
-
-						let sub_id = rand::random::<SubscriptionId>() & JS_NUM_MASK;
-
-						subscribers.lock().insert((conn, sub_id), tx.clone());
-
-						sub_id
-					};
-
-					send_response(id, tx, sub_id);
-
-					Ok(())
-				}),
-			)?;
-		}
-
-		{
-			let subscribers = subscribers.clone();
-			let result = self.insert_method(
-				unsubscribe_method_name,
-				Box::new(move |id, params, tx, conn| {
-					let sub_id = params.one()?;
-
-					subscribers.lock().remove(&(conn, sub_id));
-
-					send_response(id, tx, "Unsubscribed");
-
-					Ok(())
-				}),
-			);
-
-			match result {
-				Ok(()) => (),
-				Err(err) => {
-					// Remove subscribe method if unsubscribe has failed
-					self.methods.remove(subscribe_method_name);
-					return Err(err);
-				}
-			}
-		}
+		self.root.register_subscription(subscribe_method_name, unsubscribe_method_name)
```

Comment:
Instead of doing this proxy, we can probably add `Deref` and `DerefMut` impls to `Server` that carries over the methods for the root `RpcModule` internally.